src/async_fuse/memfs/kv_engine/xline_impl.rs

```diff
@@ -0,0 +1,365 @@
+            )
+            .await
+            .with_context(|| "failed to lock at `MetaTxn::lock`".to_owned())?;
+
+        Ok(resp.key.to_vec())
+    }
+
+    //unlock
+    async fn unlock(&self, key: Vec<u8>) -> DatenLordResult<()> {
+        let client = self.client.lock_client();
+        client
+            .unlock(UnlockRequest::new(key))
+            .await
+            .with_context(|| "failed to unlock at `MetaTxn::unlock`".to_owned())?;
+
+        Ok(())
+    }
+
+
+    /// Get the value by the key.
+    async fn get(&self, key: &KeyType) -> DatenLordResult<Option<ValueType>> {
+        let _timer = KV_METRICS.start_kv_operation_timer("get");
+        let client = self.client.kv_client();
+        let resp = client
+            .range(RangeRequest::new(key.to_string_key()))
+            .await
+            .with_context(|| format!("failed to get from etcd engine, key={key:?}"))?;
+
+        let kvs = resp.kvs.first();
+        match kvs {
+            Some(kv) => Ok(Some(serde_json::from_slice::<ValueType>(&kv.value).with_context(||{
+                "failed to deserialize value from bytes, KVEngine's value supposed to be `ValueType`".to_owned()
+            })?)),
+            None => Ok(None),
+        }
+    }
+
+
+    /// Set the value by the key.
+    async fn set(
+        &self,
+        key: &KeyType,
+        value: &ValueType,
+        _option: Option<SetOption>,
+    ) -> DatenLordResult<Option<ValueType>> {
+        let _timer = KV_METRICS.start_kv_operation_timer("set");
+        let serial_value = serde_json::to_vec(value)
+        .with_context(|| format!("failed to serialize value={value:?} to bytes"))?;
+        let mut set_request = PutRequest::new(key.to_string_key(), serial_value);
+        let _option = match _option {
+            Some(_option) => {
+                if _option.prev_kv {
+                    set_request = set_request.with_prev_kv(_option.prev_kv);
+                }
+                if let Some(lease) = _option.lease {
+                    set_request = set_request.with_lease(lease);
+                }
+                Some(&set_request)
+            }
+            None => None,
+        };
+
+        let client = self.client.kv_client();
+        let resp = client
+            .put(set_request)
+            .await
+            .with_context(|| "failed to put at `MetaTxn::set`".to_owned())?;
+        if let Some(pre_kv) = resp.prev_kv {
+            let decoded_value: ValueType = serde_json::from_slice(&pre_kv.value)?;
+            Ok(Some(decoded_value))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Delete the kv pair by the key.
+    async fn delete(
+        &self,
+        key: &KeyType,
+        _option: Option<DeleteOption>,
+    ) -> DatenLordResult<Option<ValueType>> {
+        let _timer = KV_METRICS.start_kv_operation_timer("delete");
+        let mut delete_req = DeleteRangeRequest::new(key.to_string_key());
+        let _option = match _option {
```

Comment:
Remove `_` prefix in clippy test.

Reply:
Hi @IRONICBo, since xline use request as the parameter  in `delete` and `set` function, so the `_option` will not be used. Therefore, remove the underscore followed by `option` would cause compile error. I am thinking whether I should use request as parameter type instead of using option.

Reply:
@vicluo96 Yes, the options here are mainly adapted to etcd requests, maybe consider adapting the Option type to xline requests.